Framework/PythonInterface/mantid/geometry/src/Exports/IObject.cpp

```diff
@@ -12,5 +14,7 @@ GET_POINTER_SPECIALIZATION(IObject)
 void export_IObject() {
   register_ptr_to_python<boost::shared_ptr<IObject>>();
 
-  class_<IObject, boost::noncopyable>("IObject", no_init);
+  class_<IObject, boost::noncopyable>("IObject", no_init)
```

Comment:
I don't think you need to change this file at all. CSGObject already exposes getShapeXML in mantid/Framework/PythonInterface/mantid/geometry/src/Exports/CSGObject.cpp. This allows getShapeXML to be used when a CSG object is returned from getShape. When an IObject is returned I'm not sure we want to give it a method it does not have in the underlying C++ layer. 

On the plus side when I removed all the changes to this file the unit tests and the test scripts still worked as I don't think it was doing anything. So should be good to go once the changes to this file are reverted :)

Reply:
Thank you! I shall do that now